From ca9d5bbccbba254a1b1713c5fc97686bf5ab65ac Mon Sep 17 00:00:00 2001
From: Nikola Pajkovsky <npajkovs@redhat.com>
Date: Mon, 20 Feb 2012 09:57:17 +0100
Subject: [PATCH 01/53] bugzilla: exclude appending package name for kernel
 oops

Signed-off-by: Nikola Pajkovsky <npajkovs@redhat.com>
---
 src/plugins/rhbz.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/plugins/rhbz.c b/src/plugins/rhbz.c
index 991b1cb..32b61dc 100644
--- a/src/plugins/rhbz.c
+++ b/src/plugins/rhbz.c
@@ -420,7 +420,11 @@ int rhbz_new_bug(struct abrt_xmlrpc *ax, problem_data_t *problem_data,
     }
     else
     {
-        strbuf_append_strf(buf_summary, "[abrt] %s", package);
+        if (analyzer && !strcmp(analyzer, "Kerneloops"))
+            strbuf_append_str(buf_summary, "[abrt]");
+        else
+            strbuf_append_strf(buf_summary, "[abrt] %s", package);
+
         if (function && strlen(function) < 30)
             strbuf_append_strf(buf_summary, ": %s", function);
 
-- 
1.7.11.2

