From a52c91994c4214e0629fb11c68271b3a227625fc Mon Sep 17 00:00:00 2001
From: Jiri Moskovcak <jmoskovc@redhat.com>
Date: Mon, 20 Feb 2012 16:18:37 +0100
Subject: [PATCH 02/53] debuginfo downloader - minor fix

---
 src/client-python/debuginfo.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/client-python/debuginfo.py b/src/client-python/debuginfo.py
index 634fde6..44e3d8b 100644
--- a/src/client-python/debuginfo.py
+++ b/src/client-python/debuginfo.py
@@ -156,7 +156,7 @@ class DebugInfoDownload(YumBase):
         unmute_stdout()
 
     # return value will be used as exitcode. So 0 = ok, !0 - error
-    def download(self, files, exact_files=False):
+    def download(self, files, download_exact_files=False):
         """ @files - """
         installed_size = 0
         total_pkgs = 0
@@ -291,7 +291,7 @@ class DebugInfoDownload(YumBase):
                 print (_("Downloading package {0} failed").format(pkg))
             else:
                 unpack_result = unpack_rpm(package_file_name, files, self.tmpdir,
-                                           self.cachedir, self.keeprpms, exact_files=exact_files)
+                                           self.cachedir, self.keeprpms, exact_files=download_exact_files)
                 if unpack_result == RETURN_FAILURE:
                     # recursively delete the temp dir on failure
                     print _("Unpacking failed, aborting download...")
-- 
1.7.11.2

