From b12c72458771e23e66d2057160c604c13571244b Mon Sep 17 00:00:00 2001
From: Jiri Moskovcak <jmoskovc@redhat.com>
Date: Thu, 12 Jul 2012 13:23:44 +0200
Subject: [PATCH 23/53] fixed build errors

---
 src/lib/json.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/lib/json.c b/src/lib/json.c
index 2adf4ee..40058eb 100644
--- a/src/lib/json.c
+++ b/src/lib/json.c
@@ -37,7 +37,7 @@ static int get_pd_int_item(problem_data_t *pd, const char *key, int *result)
     if (!pd || !key)
         return 0;
 
-    char *pd_item = get_problem_item_content_or_NULL(pd, key);
+    const char *pd_item = get_problem_item_content_or_NULL(pd, key);
     if (!pd_item)
     {
         VERB1 log("warning: '%s' is not an item in problem directory", key);
@@ -75,7 +75,7 @@ static void ureport_add_str(struct json_object *ur, const char *key,
 
 static void ureport_add_os(struct json_object *ur, problem_data_t *pd)
 {
-    char *pd_item = get_problem_item_content_or_NULL(pd, FILENAME_OS_RELEASE);
+    const char *pd_item = get_problem_item_content_or_NULL(pd, FILENAME_OS_RELEASE);
     if (!pd_item)
         return;
 
@@ -97,7 +97,7 @@ static void ureport_add_os(struct json_object *ur, problem_data_t *pd)
 
 static void ureport_add_type(struct json_object *ur, problem_data_t *pd)
 {
-    char *pd_item = get_problem_item_content_or_NULL(pd, FILENAME_ANALYZER);
+    const char *pd_item = get_problem_item_content_or_NULL(pd, FILENAME_ANALYZER);
     if (!pd_item)
         return;
 
@@ -111,7 +111,7 @@ static void ureport_add_type(struct json_object *ur, problem_data_t *pd)
 
 static void ureport_add_core_backtrace(struct json_object *ur, problem_data_t *pd)
 {
-    char *pd_item = get_problem_item_content_or_NULL(pd, FILENAME_CORE_BACKTRACE);
+    const char *pd_item = get_problem_item_content_or_NULL(pd, FILENAME_CORE_BACKTRACE);
     if (!pd_item)
         return;
 
@@ -163,7 +163,7 @@ static void ureport_add_core_backtrace(struct json_object *ur, problem_data_t *p
 static void ureport_add_item_str(struct json_object *ur, problem_data_t *pd,
                                  const char *key, const char *rename)
 {
-        char *pd_item = get_problem_item_content_or_NULL(pd, key);
+        const char *pd_item = get_problem_item_content_or_NULL(pd, key);
         if (!pd_item)
             return;
 
-- 
1.7.11.2

