From 83e119fd22cfcda02372f2119644e247bf4ba285 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Mon, 28 May 2012 16:05:40 +0200
Subject: [PATCH 30/53] track#480 : unused pointer value in bodhi_read_value()

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/plugins/abrt-bodhi.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/plugins/abrt-bodhi.c b/src/plugins/abrt-bodhi.c
index 3275086..91f0eed 100644
--- a/src/plugins/abrt-bodhi.c
+++ b/src/plugins/abrt-bodhi.c
@@ -161,8 +161,7 @@ static void bodhi_read_value(json_object *json, const char *item_name,
         *(int *) value = json_object_get_int(j);
         break;
     case BODHI_READ_STR:
-        *(char **) value = (char *) xstrdup(json_object_to_json_string(j));
-        value = strtrimch(*(char **) value, '"');
+        *(char **) value = (char *) strtrimch(xstrdup(json_object_to_json_string(j)), '"');
         break;
     case BODHI_READ_JSON_OBJ:
         *(json_object **) value = (json_object *) j;
-- 
1.7.11.2

