From 853a129582d694cfa12e61eb14cc50c2b836f060 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Mon, 28 May 2012 14:34:48 +0200
Subject: [PATCH 32/53] strbuf::strtrimch() : memmove() reads bytes out of
 borders

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/lib/strbuf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/strbuf.c b/src/lib/strbuf.c
index 7036272..22cd583 100644
--- a/src/lib/strbuf.c
+++ b/src/lib/strbuf.c
@@ -76,7 +76,7 @@ char *strtrimch(char *str, int ch)
     while (*tmp == ch)
         ++tmp;
 
-    memmove(str, tmp, strlen(str));
+    memmove(str, tmp, strlen(tmp));
 
     // Remove trailing spaces.
     int i = strlen(str);
-- 
1.7.11.2

