From ab55a4259db0e6361ebba8b4e1232705527f0167 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Tue, 29 May 2012 15:54:07 +0200
Subject: [PATCH 33/53] don't leak "release" variable

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
 src/lib/dump_dir.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/lib/dump_dir.c b/src/lib/dump_dir.c
index 374c641..b4aad7e 100644
--- a/src/lib/dump_dir.c
+++ b/src/lib/dump_dir.c
@@ -488,14 +488,14 @@ void dd_create_basic_files(struct dump_dir *dd, uid_t uid)
     char *release = dd_load_text_ext(dd, FILENAME_OS_RELEASE,
                     DD_FAIL_QUIETLY_ENOENT | DD_LOAD_TEXT_RETURN_NULL_ON_FAILURE);
 
-    if (release)
-        return;
-
-    release = load_text_file("/etc/system-release",
-            DD_LOAD_TEXT_RETURN_NULL_ON_FAILURE);
     if (!release)
-        release = load_text_file("/etc/redhat-release", /*flags:*/ 0);
-    dd_save_text(dd, FILENAME_OS_RELEASE, release);
+    {
+        release = load_text_file("/etc/system-release",
+                DD_LOAD_TEXT_RETURN_NULL_ON_FAILURE);
+        if (!release)
+            release = load_text_file("/etc/redhat-release", /*flags:*/ 0);
+        dd_save_text(dd, FILENAME_OS_RELEASE, release);
+    }
     free(release);
 }
 
-- 
1.7.11.2

