From d3c4f8b8158db616a432be64e5717b176de04f6c Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Tue, 29 May 2012 14:46:47 +0200
Subject: [PATCH 36/53] track#542 : add 'reporter' and 'related_packages'
 fields to ureports

Acked-by: Nikola Pajkovsky <npajkovs@redhat.com>
Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/lib/json.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/lib/json.c b/src/lib/json.c
index a44c75d..a34cc7b 100644
--- a/src/lib/json.c
+++ b/src/lib/json.c
@@ -196,6 +196,25 @@ static void ureport_add_pkg(struct json_object *ur, problem_data_t *pd)
     json_object_object_add(ur, "installed_package", jobject);
 }
 
+static void ureport_add_related_pkgs(struct json_object *ur, problem_data_t *pd)
+{
+    // TODO: populate this field
+    struct json_object *jobject = json_object_new_array();
+    json_object_object_add(ur, "related_packages", jobject);
+}
+
+static void ureport_add_reporter(struct json_object *ur, const char *name, const char *version)
+{
+    struct json_object *jobject = json_object_new_object();
+    if (!jobject)
+        die_out_of_memory();
+
+    ureport_add_str(jobject, "name", name);
+    ureport_add_str(jobject, "version", version);
+
+    json_object_object_add(ur, "reporter", jobject);
+}
+
 char *new_json_ureport(problem_data_t *pd)
 {
     struct json_object *ureport = json_object_new_object();
@@ -219,9 +238,11 @@ char *new_json_ureport(problem_data_t *pd)
     ureport_add_type(ureport, pd);
 
     ureport_add_pkg(ureport, pd);
+    ureport_add_related_pkgs(ureport, pd);
     ureport_add_os(ureport, pd);
 
     ureport_add_core_backtrace(ureport, pd);
+    ureport_add_reporter(ureport, "ABRT", VERSION);
 
     char *j = xstrdup(json_object_to_json_string(ureport));
     json_object_put(ureport);
-- 
1.7.11.2

