From 50bd09ce84e57febf5fcc619ba8141d6b7f44f73 Mon Sep 17 00:00:00 2001
From: Jiri Moskovcak <jmoskovc@redhat.com>
Date: Thu, 7 Jun 2012 11:39:04 +0200
Subject: [PATCH 41/53] use retrace instead of local gdb in ureport event

---
 src/plugins/ureport_event.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/ureport_event.conf b/src/plugins/ureport_event.conf
index 2d465b8..94cdbac 100644
--- a/src/plugins/ureport_event.conf
+++ b/src/plugins/ureport_event.conf
@@ -5,7 +5,7 @@ EVENT=report_uReport analyzer=Python reporter-ureport
 EVENT=report_uReport analyzer=Kerneloops reporter-ureport
 
 EVENT=report-gui analyzer=CCpp
-        report-gtk -e report_uReport -e analyze_LocalGDB -e report_Bugzilla -- "$DUMP_DIR"
+        report-gtk -e report_uReport -e analyze_RetraceServer -e report_Bugzilla -- "$DUMP_DIR"
 
 EVENT=report-gui analyzer!=CCpp
         report-gtk -e report_uReport -e report_Bugzilla -- "$DUMP_DIR"
-- 
1.7.11.2

