From 45165151467618dc1c2b14549ccf00126d49d95f Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Wed, 13 Jun 2012 10:56:33 +0200
Subject: [PATCH 42/53] reporter-bugzilla : get function names consistent in
 *.c and *.h files

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/plugins/rhbz.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/rhbz.h b/src/plugins/rhbz.h
index 38c6430..a945677 100644
--- a/src/plugins/rhbz.h
+++ b/src/plugins/rhbz.h
@@ -100,7 +100,7 @@ int rhbz_new_bug(struct abrt_xmlrpc *ax, problem_data_t *problem_data,
                  const char *release,
                  int depend_on_bug);
 
-int rhbz_attach_big_files(struct abrt_xmlrpc *ax, const char *bug_id,
+int rhbz_attach_files(struct abrt_xmlrpc *ax, const char *bug_id,
                      problem_data_t *problem_data, int flags);
 
 int rhbz_attach_blob(struct abrt_xmlrpc *ax, const char *filename,
-- 
1.7.11.2

