From 60505dff98ac62f0788e8b640e6e0fe579cbd42a Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Mon, 27 Aug 2012 16:58:58 +0200
Subject: [PATCH 60/62] rhbz#747410: reporter-bugzilla: do not attach empty
 files

- see trac#678

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/plugins/rhbz.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/plugins/rhbz.c b/src/plugins/rhbz.c
index b3d4a8f..5747ae1 100644
--- a/src/plugins/rhbz.c
+++ b/src/plugins/rhbz.c
@@ -776,6 +776,13 @@ int rhbz_attach_files(struct abrt_xmlrpc *ax, const char *bug_id,
         if (is_in_string_list(name, (char**)g_not_attached_files))
             continue;
 
+        /* Skip empty files */
+        if (value->content[0] == '\0')
+        {
+            log(_("Not attaching empty file '%s'"), name);
+            continue;
+        }
+
         if (value->flags & CD_FLAG_TXT)
         {
             const char *content = value->content;
-- 
1.7.11.4

