From 0290e231e3541d3caa6c6a00d47bda838b39bbeb Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <dvlasenk@redhat.com>
Date: Wed, 27 Mar 2013 13:03:03 +0100
Subject: [RHEL6 LIBREPORT PATCH 83/94] reporter-rhtsupport: fix double-free
 error

Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
Signed-off-by: Jiri Moskovcak <jmoskovc@redhat.com>
---
 src/plugins/reporter-rhtsupport.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/plugins/reporter-rhtsupport.c b/src/plugins/reporter-rhtsupport.c
index 94b0634..1a542ee 100644
--- a/src/plugins/reporter-rhtsupport.c
+++ b/src/plugins/reporter-rhtsupport.c
@@ -480,7 +480,11 @@ int main(int argc, char **argv)
             log("URL=%s", result->url);
         }
         /* else: error msg was already emitted by dd_opendir */
+
         url = result->url;
+        result->url = NULL;
+        free_rhts_result(result);
+        result = NULL;
     }
 
     /* Attach the tarball of -d DIR */
-- 
1.8.2.1

