From 7e659349958a171fa0c05a004a63d3fb75da457e Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <dvlasenk@redhat.com>
Date: Thu, 28 Mar 2013 16:08:58 +0100
Subject: [RHEL6 LIBREPORT PATCH 86/94] remove new line from ask/ask_password
 responses

Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
Signed-off-by: Jiri Moskovcak <jmoskovc@redhat.com>
---
 src/lib/client.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/lib/client.c b/src/lib/client.c
index 3dce30b..48e7f3c 100644
--- a/src/lib/client.c
+++ b/src/lib/client.c
@@ -78,7 +78,10 @@ char *ask(const char *question)
 
     fflush(stdout);
 
-    return xmalloc_fgets(stdin);
+    char *result = xmalloc_fgets(stdin);
+    strtrimch(result, '\n');
+
+    return result;
 }
 
 char *ask_password(const char *question)
@@ -91,7 +94,10 @@ char *ask_password(const char *question)
     fflush(stdout);
 
     bool changed = set_echo(false);
+
     char *result = xmalloc_fgets(stdin);
+    strtrimch(result, '\n');
+
     if (changed)
         set_echo(true);
 
-- 
1.8.2.1

