From 575be25f20261e29b47a3701500e309d81a462e2 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <dvlasenk@redhat.com>
Date: Thu, 7 Mar 2013 17:27:33 +0100
Subject: [LIBREPORT PATCH] run_event_state: expose children_count in python
 wrapper

Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
---
 src/report-python/run_event.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/report-python/run_event.c b/src/report-python/run_event.c
index 50e3794..aa7e0cf 100644
--- a/src/report-python/run_event.c
+++ b/src/report-python/run_event.c
@@ -152,6 +152,12 @@ static PyObject *get_logging_callback(PyObject *pself, void *unused)
     Py_RETURN_NONE;
 }
 
+static PyObject *get_children_count(PyObject *pself, void *unused)
+{
+    p_run_event_state *self = (p_run_event_state*)pself;
+    return PyInt_FromLong(self->state ? self->state->children_count : 0);
+}
+
 static int set_post_run_callback(PyObject *pself, PyObject *callback, void *unused)
 {
     p_run_event_state *self = (p_run_event_state*)pself;
@@ -202,6 +208,7 @@ static PyGetSetDef p_run_event_state_getset[] = {
     /* attr_name, getter_func, setter_func, doc_string, void_param */
     { (char*) "post_run_callback", get_post_run_callback, set_post_run_callback },
     { (char*) "logging_callback" , get_logging_callback , set_logging_callback  },
+    { (char*) "children_count"   , get_children_count   , NULL                  },
     { NULL }
 };
 
-- 
1.8.3.1

