From 8ec479cd5a6e8b8f304734f78d01f7f27e2df91c Mon Sep 17 00:00:00 2001
From: Matej Habrnal <mhabrnal@redhat.com>
Date: Fri, 29 Jan 2016 15:49:22 +0100
Subject: [PATCH] uploader: fix - compare tmp var before it is freed

Signed-off-by: Matej Habrnal <mhabrnal@redhat.com>
---
 src/plugins/reporter-upload.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plugins/reporter-upload.c b/src/plugins/reporter-upload.c
index 92b49b4..ba8b765 100644
--- a/src/plugins/reporter-upload.c
+++ b/src/plugins/reporter-upload.c
@@ -66,12 +66,12 @@ static int create_and_upload_archive(
     if (url && url[0] && strcmp(url, "file:///tmp/") != 0)
     {
         char *tmp = upload_file(url, tempfile, settings);
+        result = (tmp == NULL);
+
         if (remote_name)
             *remote_name = tmp;
         else
             free(tmp);
-
-        result = (tmp == NULL);
     }
     else
     {
-- 
1.7.1

