diff -ruN libreport-2.1.6/src/gui-wizard-gtk/wizard.c libreport-2.1.6.rhel7/src/gui-wizard-gtk/wizard.c
--- libreport-2.1.6/src/gui-wizard-gtk/wizard.c	2013-07-26 06:19:00.000000000 +0200
+++ libreport-2.1.6.rhel7/src/gui-wizard-gtk/wizard.c	2013-07-29 15:22:50.383899811 +0200
@@ -1702,14 +1702,7 @@
 "\t▫ <b>network connection problems</b>\n"\
 "\t▫ <b>corrupted problem data</b>\n"));
 
-    add_warning(
-_("If you want to help us, please click on the upload button and provide all problem data for a deep analysis.\n"\
-"<i>Before you do that, please consider the security risks. Problem data may contain sensitive information like passwords.\n"\
-"The uploaded data are stored in a protected storage and only a limited number of persons can read them.</i>"));
-
     show_warnings();
-
-    gtk_widget_show(g_btn_onfail);
 }
 
 static gboolean consume_cmd_output(GIOChannel *source, GIOCondition condition, gpointer data)
